### ISSUE TITLE
Relaying on substrate chain raises misbehaviour detection errors

### ISSUE THREAD

**Issue by DaviRain-Su** [informalsystems/hermes#1427]
`ERROR ThreadId(22) [ibc-0 -> ibc-1:10-grandpa-0] MISBEHAVIOUR DETECTED Misbehaviour, sending evidence`

This is the error I get when I start `cargo run -- -c config.toml start` with `relayer-cli`, and I'm forking this relayer as a relayer between substrate chains.

Another one is that misbehaviour doesn't seem to be included in the ics02 handler.

**Comment by romac** [comment]
@DaviRain-Su Any update on this?

**Comment by DaviRain-Su** [comment]
> @DaviRain-Su Any update on this?

No feedback for now, these are because of the problems I encountered in integrating relayer support for substrate. I'll give feedback when I encounter new problems.

**Comment by DaviRain-Su** [comment]
> Hi DaviRain, thanks for opening this. It would be good if we can get more information on trying to reproduce this problem. Can you offer more details on how you instantiate the substrate chains?
> 
> At the moment the relayer has no support for substrate chains, so this issue regarding misbeharior is part of a bigger problem or adding support for non-SDK chains.
Yes there are many problems to solve in integrating relayer support for substrate. For example, the mismatch between substrate chain and cosmos-sdk in terms of block speed, substrate gets data directly from the chain via subxt(https://github.com/paritytech/subxt), which is not quite the same as cosmos (here we are using tendermint-rpc).

**Comment by DaviRain-Su** [comment]
> > Hi DaviRain, thanks for opening this. It would be good if we can get more information on trying to reproduce this problem. Can you offer more details on how you instantiate the substrate chains?
>> At the moment the relayer has no support for substrate chains, so this issue regarding misbeharior is part of a bigger problem or adding support for non-SDK chains.

Yes there are many problems to solve in integrating relayer support for substrate. For example, the mismatch between substrate chain and cosmos-sdk in terms of block speed, substrate gets data directly from the chain via subxt(https://github.com/paritytech/subxt), which is not quite the same as cosmos (here we are using tendermint-rpc).
